tests/python/contrib/test_hexagon/benchmarks_table.py

```diff
@@ -0,0 +1,231 @@
+          defined for this benchmark table.
+        """
+        self.validate_user_supplied_kwargs(kwargs)
+
+        line_item = dict(kwargs)
+
+        line_item['row_status'] = 'SUCCESS'
+        line_item['timings_min_usecs'] = timings.min * 1000000
+        line_item['timings_max_usecs'] = timings.max * 1000000
+        line_item['timings_median_usecs'] = timings.median * 1000000
+
+        self.line_items_.append(line_item)
+
+    def record_skip(self, **kwargs):
+        self.validate_user_supplied_kwargs(kwargs)
+
+        line_item = dict(kwargs)
+        line_item['row_status'] = 'SKIP'
+        self.line_items_.append(line_item)
+
+    def record_fail(self, **kwargs):
+        self.validate_user_supplied_kwargs(kwargs)
+
+        line_item = dict(kwargs)
+        line_item['row_status'] = 'FAIL'
+        self.line_items_.append(line_item)
+
+    def has_fail(self):
+        """
+        Returns True if the table contains at least one 'file' line item,
+        otherwise returns False.
+        """
+        for li in self.line_items_:
+            if li['row_status'] == 'FAIL':
+                return True
+
+        return False
+
+    def print_csv(self, f, column_name_order, timing_decimal_places=3):
+        """
+        Print the benchmark results as a csv.
+
+        `f` : The output stream.
+
+        `column_name_order`: an iterable sequence of column names, indicating the
+           order of column in the CSV output.
+           Each string must be one of the column names provided by
+           BUILTIN_COLUMN_NAMES_TO_DESCS or provided to the class constructor.
+
+           The CSV output will contain only those columns that are mentioned in
+           this list.
+
+        `timing_decimal_places`: for the numeric timing values, this is the
+           number of decimal places to provide in the printed output.
+           For example, a value of 3 is equivalent to the Python formatting string
+           `'{:.3f}'`
+        """
+        csv.register_dialect(
+            "benchmarks",
+            delimiter="\t",
+            quotechar='"',
+            quoting=csv.QUOTE_MINIMAL,
+        )
+
+        output_order_cm_list = self.all_cols_metadata_.get_ordered_by_name_sequence(column_name_order)
+
+        output_order_header_texts = [ cm.header_text for cm in output_order_cm_list ]
+
+        writer = csv.DictWriter(f, output_order_header_texts, dialect="benchmarks", restval="")
+
+        writer.writeheader()
+        for line_item_dict in self.line_items_:
+            for k in [
+                    "timings_median_usecs",
+                    "timings_min_usecs",
+                    "timings_max_usecs",
+                    ]:
+                if k in line_item_dict:
+                    old_value = line_item_dict[k]
+                    if isinstance(old_value, float):
+                        str_value = f"{old_value:>0.{timing_decimal_places}f}"
+                        line_item_dict[k] = str_value
```

Comment:
> I'd recommend constructing a temporary dictionary containing the updated object.

Good idea. Done.

> We can also avoid the nested format string by using the [builtin round](https://docs.python.org/3/library/functions.html#round) function

In this particular case I prefer keeping  rounding / truncation in the string-rendering instead of using `round(...)`, for several reasons:
- Some programmers (at least me) are accustomed to being vigilant about floating-point imprecision when representing base-10 fractions.  So even if the recipe you gave works in practice, it can be distracting because it fails a smell test.
- Using only `round(...)` removes the users' ability to specify the exact number of decimal places in the output.  E.g., the number `print(round(4.2, 3))` will print "4.2", not "4.200".  This removes some context w.r.t. the precision of the measurements.

Reply:
Good point, and I'm convinced on it.